tests/test_metahyper/test_instance_mapping.py

```diff
@@ -0,0 +1,24 @@
+import pytest
+
+from metahyper.exceptions import MissingDependencyError
+from metahyper.utils import instance_from_map
+
+
+@pytest.mark.parametrize(
+    "err",
+    [
+        MissingDependencyError(
+            libname="neps",
+            dep="test_dep_1",
+            install_group="group_1",
+            cause=ImportError(),
+        ),
+        MissingDependencyError(
+            libname="neps", dep="test_dep_2", install_group=None, cause=ImportError()
+        ),
+    ],
+)
+@pytest.mark.metahyper
+def test_missing_dependancy_gets_flagged(err: MissingDependencyError) -> None:
+    with pytest.raises(MissingDependencyError, match=err.dep):
+        instance_from_map(mapping={}, request=err)
```

Comment:
[This](https://github.com/automl/neps/blob/f6cecbf6ae410fa45dc9a5e71424e88b3adfc4f0/src/neps/optimizers/bayesian_optimization/optimizer.py#L156-L161) is how we usually call `instance_from_map(...)` function to get the surrogate model, the `surrogate_model` parameter is usually a string

Reply:
Thanks for the reference, super useful!